galsim/interpolant.py

```diff
@@ -601,6 +624,78 @@ def krange(self):
         return 1.6058208066649935 / self._gsparams.kvalue_accuracy**(1./3.)
 
 
+class QuinticBis(Interpolant):
+    """Fifth order interpolation
+
+    The quintic interpolant is an improved version over the
```

Comment:
I agree "improved" is a always wrt some point of view. 
I will answer to what are the differences.

Reply:
Here are some explanations how I've found the expression of the QuinticBis kernel.
Let us define a piecewise polyniomal kernel as followed
```math
K(x) = \left\{ \begin{array}{ll}
\displaystyle
\sum_{j=0}^n a_{ ji } |x|^j & i \leq |x| < i+1 \\
0 & |x| \geq m 
\end{array} \right.
```
with $n=2m-1$ (m=2 => cubic, m=3 => quintic). There are $2m^2$ coefficients.

Here are the constraints that one can consider in the real-space:
```math
(1) \quad K(j) = \delta^K_{j0},\ \forall j\in \{0..m-1\}
``` 
```math
(2) \quad a_{\ell,0}=0, \quad \ell \in  \{1..m-1\}\ \mathrm{and} \ \ell \ \mathrm{odd}
```
and for $m \geq 3$:
```math
(3) \quad \lim_{x\rightarrow j^{-}}K^{(\ell)}(x) = \lim_{x\rightarrow j^{+}}K^{(\ell)}(x), \forall j\in \{1..m\} \ \mathrm{and}\ \forall \ell \in  \{0..n-4\}
```
Now, considering constraints in the Fourier space:
```math
(4)\quad \widehat{K}(0)=1
```
```math
(5)\quad \widehat{K}^{(\ell)}(0) = 0, \forall \ell\in \{2..n\} \ \mathrm{and}\ \ \ell \ \mathrm{even}
``` 
```math
(6)\quad \widehat{K}^{(\ell)}(1) = 0, \forall \ell\in \{0..n-1\}
```
It turns out that one recover for $m=2$ the usual cubic kernel, there is also a septic kernel (not considered later). Concerning the Quintic case, ie. $m=3$, there is a coefficient left non determined (a degree of freedom), one can choose $a_{20}$ for instance. How to choose this coeff?

Well, the Taylor development of $K(x)$ at $x=0$ leads to
```math
K(x) = 1 + a_{20} x^2 + O(x^3)
```
If one chooses $a_{20}=0$ to "flatten" the behaviour around x=0 then one recovers Gary's quintic kernel (nicknamed hereafter $K^G$). With such definition
```math
 K^{G}(x)~\frac{1}{12} |x|^3 + O(x^4)
```
while concerning the Fourier space
```math
\begin{array}{rcl}
\widehat{K}^G(u) &=& 1 - \frac{25 \pi^6}{189} u^6 + O(u^7) \\
\widehat{K}^G(u) &=& 2(\pi^2-27) (u-1)^5 + O((u-1)^6)
\end{array}
```
Looking at the difference of behaviours around $u=0$ and $u=1$, and remembering the law that governs the 1st ghost intensity, I've made an another    choice:
```math
a_{2,0}\to -\frac{5 \left(27-\pi ^2\right)}{4 \left(15-\pi ^2\right)}
```
leading to 
```math
\begin{array}{rcl}
\widehat{K}^{JE}(u) &=& 1-\frac{\left(\pi ^6 \left(143 \pi ^2-2361\right)\right) }{945 \left(\pi ^2-15\right)}u^6+O\left(u^7\right)\\
\widehat{K}^{JE}(u) &=& \left(-71-7 \pi ^2-\frac{1080}{\pi ^2-15}\right) (u-1)^6+O\left((u-1)^7\right)
\end{array}
```

So, to summarise
1. Both kernels share the "quintic" property of iexact interpolation of fourth-order polynomial functions, and satisfies the partition of unity both in real ans Fourier spaces.
2.  $K^G(x)$ is more flat than $K^{JE}(x)$ around $x=0$:
```math
\begin{array}{rcl}
K^{G}(x) &\approx& 1. - 7.91667 |x|^3 \\
K^{JE}(x) &\approx& 1. -4.17375 x^2
\end{array}
```
3.  $\widehat{K}^G(u)$ is a little more flat than $\widehat{K}^{JE}(u)$ around $u=0$:
```math
\begin{array}{rcl}
\widehat{K}^{G}(u) &\approx& 1.-127.168 u^6 \\
\widehat{K}^{JE}(u) &\approx&1.-188.312 u^6
\end{array}
```
4. $\widehat{K}^{JE}(u)$ is more flat than $\widehat{K}^G(u)$ around $u=1$ ($j=1$ ghost):
```math
\begin{array}{rcl}
\widehat{K}^{G}(u) &\approx& -34.2608 (u-1.)^5 \\
\widehat{K}^{JE}(u) &\approx& 70.4229 (u-1.)^6
\end{array}
```
5. $\widehat{K}^{JE}(u)$ has higher $j>1$ ghosts than $\widehat{K}^{G}(u)$ has a matter of compensation of the 1st ghost intensity. This is illustrated below by the absolute value of the coeff. of the expension of $\hat{K}(u)$ around $u=j$. 
![image](https://github.com/GalSim-developers/GalSim/assets/20539759/d364f0b1-c114-466f-a13f-afd1d85b756f)

To get a visual inspection of 1st ghost comparison as Fig 1 of Gary's 2014 paper
![image](https://github.com/GalSim-developers/GalSim/assets/20539759/57b59e65-2f8f-4858-8c83-ee43571d1763)
and concerning a Shear exercise as the one of Gary's paper
![image](https://github.com/GalSim-developers/GalSim/assets/20539759/34615388-1d52-4c2a-893a-33deb7051963)

Reply:
I have written a summary on the comment of the QinticBis class in the `interpolant.py` file. But I'm not sure of the rendering of the Latex formula.

Reply:
I'm trying to recover `test_interpolants.py`.

Reply:
I've succeeded to run w/o crashes
```
 python test_interpolants_parametric.py
```
with `use_interpolants = ['quintic','quinticbis',]` with x4 and x6 padding only, has it takes too long to get all interpolants, and I guess we wnat to focus on the differences btw both quintic kernels.
I got then two files

```
interpolant_test_parametric_output_delta.dat
interpolant_test_parametric_output_original.dat
```
which are full of digits, eg.
```
0 1 7 4 0.0 0.0 1.0 0.0 0.0 0.0 0.25770524553031665 0.25770524553031665 -3.744121318538518e-08 -0.1704341300636095 -0.1704341300636095 3.1705050529406975e-09 12.274123191833496 12.274123191833496 0.0
1 1 7 4 0.0 0.0 1.0 0.0 0.0 0.0 0.17104476123913895 0.17104476123913895 0.0 -0.08506195174953833 -0.08506195174953833 0.0 24.92190933227539 24.92190933227539 0.0
2 1 7 4 0.0 0.0 1.0 0.0 0.0 0.0 -0.11189117548202691 -0.11189117548202691 6.33555060775004e-08 0.11695983838376801 0.11695983838376801 -2.522577972896567e-08 9.526039123535156 9.526039123535156 2.0022473223946557e-07
3 1 7 4 0.0 0.0 1.0 0.0 0.0 0.0 -0.11641141490902396 -0.11641141490902396 1.1733047446116363e-07 -0.37654191463949516 -0.37654191463949516 1.4869379583171138e-07 8.769744873046875 8.769744873046875 3.2623787700049066e-07
4 1 7 4 0.0 0.0 1.0 0.0 0.0 0.0 0.08082227985345199 0.08082227985345199 -3.4843277755311486e-05 -0.1910075380407282 -0.1910075380407282 2.510564681271199e-05 3.2600162029266357 3.2600162029266357 0.00021194282458352468
5 1 7 4 0.0 0.0 1.0 0.0 0.0 0.0 -0.17088581813656703 -0.17088581813656703 1.633058779870744e-08 0.06538689404139647 0.06538689404139647 -3.560703398797216e-10 15.034000396728516 15.034000396728516 6.343450121324828e-08
6 1 7 4 0.0 0.0 1.0 0.0 0.0 0.0 0.08044154702195001 0.08044154702195001 0.0 0.004146179231359682 0.004146179231359682 0.0 29.20765495300293 29.20765495300293 0.0
7 1 7 4 0.0 0.0 1.0 0.0 0.0 0.0 -0.2006712890480676 -0.2006712890480676 1.743652874863777e-08 0.11759667667616072 0.11759667667616072 -8.867007947332972e-09 14.169123649597168 14.169123649597168 0.0
8 1 7 4 0.0 0.0 1.0 0.0 0.0 0.0 -10.0 -10.0 0.0 -10.0 -10.0 0.0 -10.0 -10.0 -0.0
9 1 7 4 0.0 0.0 1.0 0.0 0.0 0.0 -0.26816091307716905 -0.26816091307716905 7.530807773392656e-05 -0.2307710341157043 -0.2307710341157043 2.6944817052981795e-05 3.7241814136505127 3.7241814136505127 0.00024256820382182243
```
I've seen then `plot_test_interpolants.py` which I have tuned blindly 
```
interpolant_titles = ['quintic','quinticbis','default']
interpolant_colors = ['#000000', '#db00db', '#f80000']
padding_titles = ['pad4','pad6']
padding_shapes = ['o', 's']
```
With the "delta" output file I got many PNG files but get a crash 
```
Traceback (most recent call last):
  File "/sps/lsst/users/campagne/GalSim-r2.5-forked/devel/modules/plot_test_interpolants.py", line 319, in <module>
    plot_interpolants(sys.argv[-2], sys.argv[-1])
  File "/sps/lsst/users/campagne/GalSim-r2.5-forked/devel/modules/plot_test_interpolants.py", line 289, in plot_interpolants
    plt.errorbar(bins[k]+ixoffset*xoffset, mean, yerr=[q1,q3],
  File "/pbs/home/c/campagne/my_sps/miniconda3/envs/galsim/lib/python3.11/site-packages/matplotlib/pyplot.py", line 3027, in errorbar
    return gca().errorbar(
           ^^^^^^^^^^^^^^^
  File "/pbs/home/c/campagne/my_sps/miniconda3/envs/galsim/lib/python3.11/site-packages/matplotlib/__init__.py", line 1465, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/pbs/home/c/campagne/my_sps/miniconda3/envs/galsim/lib/python3.11/site-packages/matplotlib/axes/_axes.py", line 3674, in errorbar
    raise ValueError(
ValueError: 'yerr' must not contain negative values
```
Now, the PNG are plots but with legend 'quinticbis' & 'default', and w/o any "quintic" points. So it might be that the settings 
```
interpolant_titles = ['quintic','quinticbis','default']
interpolant_colors = ['#000000', '#db00db', '#f80000']
```
are not correct. Now the "original" dat file produce errors, and no PNG.

So, if someone can remind who has done these codes and how to adapt to the current API, it would be great.

Reply:
hi again, debugging lead that the interpolant indexes are`[1, 2, 7]` so tweeking a bit 
```
interpolant_titles = ['dummy','quintic','quinticbis'
                          ,'dummy','dummy','dummy','default']
```
leads removing the yerr crash. I just get 
```
Unable to parse the pattern
/usr/lib64/python2.7/site-packages/matplotlib/axes.py:4602: UserWarning: No labeled objects found. Use label='...' kwarg on individual plots.
  warnings.warn("No labeled objects found. "
```
Now looking at some PNG seems to get quintic and quinticbis marker but it is quite blind analysis and I not sure to get valuable results. Now the results look very similar.